### ISSUE TITLE
Localhost doesn't work in Jolokia URL definition

### ISSUE THREAD

**Issue by wawrzek** [logzio/jmx2graphite#3]
I notice following problem.

Setting 'localhost' in URL for Jolokia in jmx2grahpite I see following error:
`
16/03/11 12:39:02 WARN i.l.j.MetricsPoller: Failed polling metrics from Jolokia: Failed retrieving list of beans from Jolokia. Error = Connect to localhost:31338 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused`

The port is definitely open, because I can get URL content via **telnet localhost 31338**.
After replacing **localhost** with **127.0.0.1**
 `
16/03/11 12:54:28 WARN i.l.j.MetricsPoller: Failed polling metrics from Jolokia: Failed retrieving list of beans from Jolokia. Error = Connect to 127.0.0.1:31338 [/127.0.0.1] failed: Connection refused`

Finally using IP address (in my case it was **10.0.1.4**) make it works, so I'm fine at the moment.

I'm using Docker as suggested at the moment on Ubuntu 14.04 on Azure. Any idea what's wrong? I think it might be Docker thing. One of the obvious test I'm planning to do is not to use Docker, but it might need to wait a bit.

**Comment by asafm** [comment]
It was a pitfall for me as well which originates in the way Docker works. Jmx2graphite runs inside Docker, which means that for it, localhost is the "mini" machine Docker created for it. The only way, I know of, to reach another process running the host machine (the ones which runs jmx2graphite docker) is to the use the host machine IP. 
The same solution applies when you poll metrics from another docker on the host machine. For example, I have a web-service wrapped in docker running on my host called foobar. I also installed and run jmx2graphite docker. When running web-service docker I specified "-v 11000" which is my jolokia agent chosen port. In the jmx2graphite I specified the IP of foobar and 11000 as the jolokia URL.

Does that make sense?
